Lesson 17/task-2.py

```diff
@@ -0,0 +1,81 @@
+class Author:
+    def __init__(self, name, country, birthday):
+        self.name = name
+        self.country = country
+        self.birthday = birthday
+        self.books = []
+
+    def __repr__(self):
+        return f'name:{self.name}, country:{self.country}, birthday:{self.birthday}, books:{len(self.books)}'
+
+    def __str__(self):
+        return f'{self.name} was born on {self.birthday}, {self.country}. Write {len(self.books)} book(s).'
+
+    def add_book(self, book):
+        if book not in self.books:
+            self.books.append(book)
+
+
+class Book:
+    total = 0
+
+    def __init__(self, name, year, author):
+        self.id = Book.total
+        self.name = name
+        self.year = year
+        self.author = author
+
+        Book.total += 1
+
+    def __repr__(self):
+        return f'id:{self.id}, name:{self.name}, year:{self.year}, author:{self.author.name}'
+
+    def __str__(self):
+        return f'{self.name} writen by {self.name} and published in {self.year}.'
+
+
+class Library:
+    def __init__(self, name):
+        self.name = name
+        self.books = []
```

Comment:
I'd combine these two into a dictionary, this would simplify the code that follows.